docs/transformer.md

```diff
@@ -0,0 +1,84 @@
+
+Let us extend our [`sayHello` example](rpc.md) with a piece of configuration which states the word to use when greeting:
+
+```haskell
+import Control.Monad.Reader
+
+sayHello :: (MonadServer m, MonadReader T.Text m)
+         => HelloRequest -> m HelloResponse
+sayHello (HelloRequest nm) = do
+  greeting <- ask
+  return (HelloResponse (greeting <> ", " <> nm))
+```
+
+Unfortunately, the simple way to run a gRPC application no longer works:
+
+```haskell
+main = runGRpcApp 8080 "helloworld" quickstartServer
+```
+
+Furthermore, how does the server know which is the actual value? In other words, how do we inject the value for `greeting`? We need to declare how to *handle* that capability. This is traditionally done with a `run` function; this additional argument is used by `runGRpcAppTrans`.
+
+```haskell
+main = runGRpcAppTrans 8080 (flip runReaderT "hi") quickstartServer
+```
+
+## Logging
+
+There are quite a number of libraries which provide logging support. Let's begin with [`monad-logger`](https://github.com/snoyberg/monad-logger#readme). In this case, an additional [set of functions](http://hackage.haskell.org/package/monad-logger/docs/Control-Monad-Logger.html#g:8) is available when you implement the `MonadLogger` class. For example, we could log a message every time we say hi:
+
+```haskell
+import Control.Monad.Logger
+
+sayHello :: (MonadServer m, MonadLogger m)
+         => HelloRequest -> m HelloResponse
+sayHello (HelloRequest nm) = do
+  logInfoN "running hi"
+  return (HelloResponse ("hi, " <> nm))
+```
+
+The most important addition with respect to the [original code](rpc.md) is in the signature. Before we only had `MonadServer m`, now we have an additional `MonadLogger m` there.
+
+As we have done with the Reader example, we need to define how to handle `MonadLogger`. `monad-logger` provides [three different monad transformers](http://hackage.haskell.org/package/monad-logger-0.3.31/docs/Control-Monad-Logger.html#g:3), so you can choose whether your logging will be completely ignored, will become a Haskell value, or would fire some `IO` action like printing in the console. Each of these monad transformers comes with a `run` action which declares how to handle it; the extended function `runGRpcAppTrans` takes that handler as argument.
+
+```haskell
+main = runGRpcAppTrans 8080 runStderrLoggingT quickstartServer
+```
+
+If you prefer other logging library, this is fine with us! Replacing `monad-logger` by [`co-log`](https://github.com/kowainik/co-log) means asking for a different capability in the server. In this case we have to declare the type of the log items as part of the `WithLog` constraint:
+
+```haskell
+import Colog.Monad
+
+sayHello :: (MonadServer m, WithLog env String m)
+         => HelloRequest -> m HelloResponse
+sayHello (HelloRequest nm) = do
+  logInfoN "running hi"
```

Comment:
I think this needs to be `logDebug` or something like that from the `co-log` library? 🤔